### DISCUSSION TITLE
How can one enable freemium plans?

### DISCUSSION THREAD

**Discussion by techwraith** [usebasejump/basejump#84] [Q&A] [answered]
Our product has a Free version that users can use for free, a Pro version that costs $25/mo (or $250/year), and an Enterprise version where they can work with us to define a yearly contract.

Basejump wants a plan ID by default, so how do we configure it so that:
1) Personal accounts don't have billing
2) Team accounts default to the free plan
3) The billing page on the team accounts allows for the user to upgrade to Pro self-serve
4) Enterprise accounts can be tracked using Stripe

I'm struggling to figure out how to make this combo happen, but I assume this is a very common use case. Any ideas?

**Comment by techwraith** [comment]
Additional details: I'm using the https://github.com/usebasejump/basejump-next starter template

**Comment by techwraith** [comment]
@tiniscule Any input here? I'm really not sure how to do this.

**Comment by tiniscule** [comment, marked as answer]
Hey there - happy to help here.

1) Personal accounts don't have billing
   - To accomplish this, just flip enable_personal_account_billing to false in basejump.config table
 
2) Team accounts default to the free plan
   - Create a free product in Stripe, take the price ID for that product and set it as the default product in Basejump env vars for the billing edge function

3) The billing page on the team accounts allows for the user to upgrade to Pro self-serve
  - Create a pro product in Stripe with the appropriate pricing for monthly/yearly.  If you configure Stripe customer portal to let people change their plans you can leverage their interface for selecting / changing plans between Free and Pro.  Just make sure you add both plans as available options to the customer portal

4) Enterprise accounts can be tracked using Stripe
  - When a user signs up for an account, Basejump automatically adds a basejump_account_id meta field to the customer and subscription for their account.  When you work out enterprise deals, just update the subscription on the customer to the appropriate custom pricing and it'll be automatically reflected on the billing page as long as that basejump_account_id remains on the subscription.  If you find yourself in a situation where a customer or subscription does not yet exist, create the subscription in stripe like you normally would and then add a metadata field called `basejump_account_id` with the users account ID as the value.  It'll then sync into your database the next time the user loads the billing page (or immediately if you have webhooks configured)

**Reply by techwraith** [reply]
Thanks @tiniscule! I've done steps 1+2, but I'm not seeing default free plans set up for teams. Is there something more I need to do?

**Reply by tiniscule** [reply]
Basejump creates customers / subscriptions in stripe when the billing endpoint is hit. If you view the billing page for a team, it should create one then.

I need to add a global component to that next starter that adds a payment banner for delinquent accounts - that would create it everywhere in the dashboard right away.

It wont' automatically create subscriptions for existing teams until they next login
